File: app/src/main/java/com/compuspulse/viewmodel/HistoryViewModel.java
package com.compuspulse.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.compuspulse.data.local.entity.HistoryArticleEntity;
import com.compuspulse.data.repository.ArticleLocalRepository;

import java.util.List;

public class HistoryViewModel extends ViewModel {

    private final ArticleLocalRepository repository = new ArticleLocalRepository();
    private final LiveData<List<HistoryArticleEntity>> historyList = repository.observeHistory();

    public LiveData<List<HistoryArticleEntity>> getHistoryList() {
        return historyList;
    }

    public void clearHistory() {
        repository.clearHistory();
    }
}

File: app/src/main/java/com/compuspulse/viewmodel/HistoryViewModel.kt
package com.compuspulse.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.compuspulse.data.local.entity.HistoryArticleEntity
import com.compuspulse.data.repository.ArticleLocalRepository
import kotlinx.coroutines.launch

class HistoryViewModel : ViewModel() {

    private val repository = ArticleLocalRepository()
    private val historyList: LiveData<List<HistoryArticleEntity>> = repository.observeHistory()

    fun getHistoryList(): LiveData<List<HistoryArticleEntity>> = historyList

    fun clearHistory() {
        viewModelScope.launch {
            repository.clearHistory()
        }
    }
}
